mono/metadata/sre.c

```diff
@@ -2350,14 +2350,36 @@ mono_reflection_get_custom_attrs_blob_checked (MonoReflectionAssembly *assembly,
 static gboolean
 reflection_setup_internal_class (MonoReflectionTypeBuilderHandle ref_tb, MonoError *error)
 {
+	static GHashTable *seen = NULL;
```

Comment:
Making this static only works as long as we don't change anything in locking.

This is a bad design.